### PR TITLE
add types!

### PR DESCRIPTION
**Pull request by yogurtandjam**
converts everything to typescript.

had to add an override in the package.json to get typescript 5.2.2 to install since react-scripts depends on typescript 3.x

### CODE REVIEW

**Review by NameFILIP — APPROVED** [review]
looks good!

File: src/Search.tsx [hunk 1]
```diff
@@ -1,12 +1,20 @@
 import debounce from "lodash.debounce";
-import { useEffect, useCallback, useState } from "react";
+import { useEffect, useCallback, useState, ChangeEvent } from "react";
 import { OBJ_URL, SEARCH_URL } from "./consts";
 import { Art } from "./models/Art";
 import Image from "./Image";
+type TArtworkIds = {
```
Comment (NameFILIP) [inline src/Search.tsx:6]: I've also seen some people use T suffix.. although I don't think its really needed at all

File: src/tf.tsx [hunk 2]
```diff
@@ -1,15 +1,30 @@
+import { FaceLandmarksDetector } from "@tensorflow-models/face-landmarks-detection";
```
Comment (NameFILIP) [inline src/tf.tsx:1]: `import type { FaceLandmarksDetector } ...`?

File: src/Search.tsx (hunk 1 above)
Comment (yogurtandjam) [reply src/Search.tsx:6]: yeah i normally don't really like it but i was having a namespace clash with the Art model. i actually think my type naming here in general could be rethought, but i figure we'll get there when we get there...

File: src/tf.tsx (hunk 2 above)
Comment (yogurtandjam) [reply src/tf.tsx:1]: oh, is that the proper way to do it? i'll update it!

**Pull request merged**